ui/src/app/views/dataset-page.spec.tsx

```diff
@@ -70,18 +70,50 @@ describe('DataSet', () => {
     expect(wrapper.find('[data-test-id="value-list-items"]').length).toBe(5);
   });
 
-  it('should enable save button once cohorts, concepts and values are selected', async() => {
-    const wrapper = mount(<DataSetPage />);
-    await waitOneTickAndUpdate(wrapper);
-    await waitOneTickAndUpdate(wrapper);
+  it('should enable save button and preview button once cohorts, concepts and values are selected',
+    async() => {
+      const wrapper = mount(<DataSetPage />);
+      await waitOneTickAndUpdate(wrapper);
+      await waitOneTickAndUpdate(wrapper);
 
-    // Preview Button by default should be disabled
-    const saveButton = wrapper.find(Button).find('[data-test-id="save-button"]')
+      // Preview Button and Save Button should be disabled by default
+      const saveButton = wrapper.find(Button).find('[data-test-id="save-button"]')
+        .first();
+      expect(saveButton.prop('disabled')).toBeTruthy();
+      const previewButton = wrapper.find(Button).find('[data-test-id="preview-button"]')
         .first();
-    expect(saveButton.prop('disabled')).toBeTruthy();
+      expect(previewButton.prop('disabled')).toBeTruthy();
+
+      // After all cohort concept and values are selected all the buttons will be enabled
+
+      wrapper.find('[data-test-id="cohort-list-item"]').first()
+        .find('input').first().simulate('change');
+      wrapper.update();
+
+      wrapper.find('[data-test-id="concept-set-list-item"]').first()
+        .find('input').first().simulate('change');
 
-    // After all cohort concept and values are selected all the buttons will be enabled
+      await waitOneTickAndUpdate(wrapper);
+
+      wrapper.find('[data-test-id="value-list-items"]').find('input').first()
+        .simulate('change');
+
+      // Buttons should now be enabled
+      const buttons = wrapper.find(Button);
+      expect(buttons.find('[data-test-id="save-button"]').first().prop('disabled'))
+        .toBeFalsy();
+      expect(buttons.find('[data-test-id="preview-button"]').first().prop('disabled'))
+        .toBeFalsy();
+    });
+
+  it('should display preview data table only once cohort, concept and value are selected ' +
```

Comment:
Sorry i did not see this in last review however, since we have test to say **preview button is enabled once cohort concept and values are selected**, in this case we can just have the description: Should display preview data table once preview button is clicked.